libraries/botbuilder-core/tests/testAdapter.test.js

```diff
@@ -496,4 +496,19 @@ describe(`TestAdapter`, function () {
         adapter.addUserToken('myConnection2', 'test', 'user', 'def456');
         adapter.send('hi');
     });
+
+    it(`should return statuses from getTokenStatus`, function (done) {
+        const adapter = new TestAdapter((context) => {
+            context.adapter.getTokenStatus(context, 'user').then(statuses => {
+                assert(statuses);
+                assert(statuses.length == 2);
+                assert(statuses.reduce((j, status) => (j||status.ConnectionName === "ABC"), false));
+                done();
```

Comment:
If an error is thrown in `.then()` in the `logic` it'll give an unhelpful `done not called` error and an uncaughtpromiseexception.

Since [`logic`](https://github.com/microsoft/botbuilder-js/blob/c39c7744eaf1bc82e897789202420659a03c460f/libraries/botbuilder-core/src/testAdapter.ts#L101) returns `Promise<void>`, can you rewrite the `logic` using async-await syntax?  You could still use `done` in combination with a try-catch to make it easier to parse. calling done with an error will cause the test to fail.